### ISSUE TITLE
`ImageBackend.AUTODETECT` does not close file handlers

### ISSUE THREAD

**Issue by YoniSchirris** [NKI-AI/dlup#151]
**Describe the bug**
`dlup.experimental_backends.autodetect_backend` opens files to see if they are openable, but the files are not closed.

**To Reproduce**
`git clone dlup && git cd dlup && git checkout limit-bounds && pip install -e .`

```py
def main():
    
    MRXS_ROOT_FOLDER = ""
    TCGA_ROOT_FOLDER = ""

    mrxs_paths = [path for path in Path(MRXS_ROOT_FOLDER).glob("*/*/*.mrxs)][:5]
    tcga_paths = [path for path in Path(TCGA_ROOT_FOLDER).glob('*/*.svs')][:5]
    for slide_path in mrxs_paths + tcga_paths:
        print(f"============================\n   Slide: {slide_path.stem}")
        for backend in [ImageBackend.OPENSLIDE, ImageBackend.PYVIPS, ImageBackend.AUTODETECT]:
            print(f"-------\n  Using backend: {backend}")
            slide = SlideImage.from_file_path(slide_path, backend=backend)
            print(f"Slide bounds: {slide.slide_bounds}")
            print(f"Full slide size: {slide.size}")
            ds = partial(TiledROIsSlideImageDataset.from_standard_tiling, path=slide_path, mpp=16, tile_size=(256, 256), tile_overlap=(0, 0), backend=backend)
            print(f"Len of tiled dataset without limit bounds: {len(ds(limit_bounds=False))}")
            print(f"Len of tiled dataset with limit bounds: {len(ds(limit_bounds=True))}")
           # ^--- this will fail only with AutoDetect backend

if __name__ == "__main__":
    main()
```
**Expected behavior**
File handler should be properly closed. `AUTODETECT` should not leave any file handlers open, and should function exactly the same as using `OPENSLIDE` or `PYVIPS` or `TIFFILE` backend.

**Environment**
dlup version: 0.3.23
How installed: git install
Python version: 3.10
Operating System: linux

**Comment by jonasteuwen** [comment]
Likely fixed by #09e638d. Can you check?

**Comment by jonasteuwen** [comment]
@YoniSchirris Can you check if this is still the case?
